Diagnostics/HealthChecker/Analyzer/Invoke-AnalyzerFrequentConfigurationIssues.ps1

```diff
@@ -315,12 +321,17 @@ function Invoke-AnalyzerFrequentConfigurationIssues {
     $showMoreInfo = $false
 
     foreach ($connector in $exoConnector) {
-        # Misconfigured connector is if TLSCertificateName is not set or CloudServicesMailEnabled not set to true
-        if ($connector.CloudEnabled -eq $false -or
-            $connector.CertificateDetails.TlsCertificateNameStatus -eq "TlsCertificateNameEmpty") {
+
+        # If CloudServiceMailEnabled is not set to true it means the connector is misconfigured
+        # If no Fqdn is set on the connector, the Fqdn of the computer is used to perform best matching certificate selection
+        # There is a risk that the Fqdn is an internal one (e.g., server.contoso.local) which will lead to a broken hybrid mail flow in case that TlsCertificateName is not set
+        if (($connector.CloudEnabled -eq $false) -or
+            ($null -eq $connector.Fqdn -and
+            $connector.CertificateDetails.TlsCertificateNameStatus -eq "TlsCertificateNameEmpty")) {
```

Comment:
@richfaj @stanalek does that work to make the check more accurate? We show a red error in case that `CloudServiceMailEnabled` is `false` or if no `Fqdn` is set on the connector => `Fqdn` of the computer will be used which is quite often an internal one (e.g., computer.domain.local) and `TlsCertificateName` is also not set => implicit matching will be performed.